Broaden the specification's scope to embedded systems with simpler hardware and reduced functionalities (#75)
riscv-non-isa/riscv-ap-tee: specification/refarch.adoc

page tables (PT) can be used to enforce isolation between TVMs and
ordinary VMs. In this model the host VMM must execute in the de-privileged VS
mode and the TSM must provide nested virtualization of the H-extension controls.
This model may be suitable for client/embedded systems and is shown in <<dep2>>.

. The TSM runs along the TSM-driver in M-mode, thus allowing only for a single confidential 
supervisor domain. This model enforces isolation between TVMs using MMU (G-stage PT) and 
between hosting supervisor domain (OS/VMM, VMs) and confidential supervisor domain using, e.g., PMPs.
This model might be suitable for client/embedded systems running in a constraint hardware and software
environment (e.g., it does not require Smmtt). <<dep3>> discusses this model in detail. 

=== CoVE Memory Isolation
Memory isolation for TVMs is orchestrated by the TSM-driver and the TSM in two
phases: the conversion of memory to confidential memory and the assignment of
confidential memory (alongwith the enforcement of properties on use) to TVMs.
To enforce isolation across Host and Confidential supervisor domains, CoVE
requires isolation of physical memory (that supports paging when enabled). 

> CoVE defines two types of memory:

Reviewer: ```suggestion
CoVE defines two types of memory:

```

Reviewer: need a line to render correctly

* Confidential memory - used for TVM/TSM code and security-sensitive data;
including state such as 1st-stage, G-stage page tables.
* Non-confidential memory - used only for shared data, e.g., communication
between the TVM/TSM and the non-TCB host software and/or non-TCB IO devices.

The split of memory into confidential and non-confidential may be static or dynamic. 
Static partitioning occurs during platform initialization while the dynamic partitioning 

> occurs during runtime. The dynamic partitioning allows for better resource utilization 
> at the cost of more complex hardware and software implementation. 

Reviewer: ```suggestion
occurs during runtime. The dynamic partitioning allows for better resource utilization, 
but has additional hardware and software requirements. 
```
